lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb

```diff
@@ -158,23 +164,27 @@ def confirm(inputmap)
         #   centos: vdb1 already mounted at /data
         if vdb1
           # TODO: check that vdb1 is mounted at /data, for now assume it is
-          log.debug 'Assuming /dev/vdb1 is mounted at /data, if this is not the case, file an issue'
+          log.debug 'Assuming /dev/vdb1 is mounted at /data or at /mnt, if this is not the case, file an issue'
         elsif vdb
           begin
-            mounted = false
-            ssh_exec!(ssh, 'if grep "vdb /data " /proc/mounts ; then /bin/false ; fi', 'Checking if /dev/vdb mounted already')
+            data_mounted = false
+            ssh_exec!(ssh, 'if grep "vdb /data " /proc/mounts ; then echo "/dev/vdb is mounted"; else /bin/false ; fi', 'Checking in /proc/mounts whether /dev/vdb mounted already')
+            data_mounted = true
           rescue
-            mounted = true
+            log.debug 'Disk /dev/vdb is not mounted to /data'
+            log.debug "data_mounted = #{data_mounted}"
           end
-          # If mounted = true, we're done
-          unless mounted
-            # disk isn't mounted at /data, could be mounted elsewhere
+          # If data_mounted = true, we're done
+          unless data_mounted
+            # disk isn't mounted at /data, could be mounted elsewhere (e.g. /mnt)
             begin
-              # Are we mounted?
-              ssh_exec!(ssh, 'mount | grep ^/dev/vdb 2>&1 >/dev/null', 'Checking if /dev/vdb is unmounted')
+              # Are we mounted? # if we are, set unmount to true
+              ssh_exec!(ssh, 'if mount | grep "^/dev/vdb " ; then echo "Disk /dev/vdb is mounted" ; else /bin/false ; fi', 'Confirm /dev/vdb is mounted')
               unmount = true
             rescue
+              mounted = false
               log.debug 'Disk /dev/vdb is not mounted'
+              unmount = false
```

Comment:
move this to just before the ssh_exec!.  for consistency with ``data_mounted`` above, and its good to minimize logic in a rescue block